editor/src/core/model/storyboard-utils.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  addFileToProjectContents,
+  getContentsTreeFileFromString,
+  walkContentsTree,
+} from '../../components/assets'
+import { EditorModel } from '../../components/editor/action-types'
+import { EditorState } from '../../components/editor/store/editor-state'
+import { forEachRight, left, right } from '../shared/either'
+import {
+  isUtopiaJSXComponent,
+  UtopiaJSXComponent,
+  utopiaJSXComponent,
+} from '../shared/element-template'
+import { importAlias, isUIJSFile, RevisionsState } from '../shared/project-file-types'
+import { generateUID } from '../shared/uid-utils'
+import { addImport, parseSuccess } from '../workers/common/project-file-utils'
+import { uiJsFile } from './project-file-utils'
+import {
+  BakedInStoryboardUID,
+  BakedInStoryboardVariableName,
+  createSceneFromComponent,
+  createStoryboardElement,
+} from './scene-utils'
+
+export const StoryboardFilePath: string = '/src/storyboard.js'
+
+const PossiblyMainComponentNames: Array<string> = ['App', 'Application', 'Main']
+
+interface ComponentToImport {
+  path: string
+  component: UtopiaJSXComponent
+}
+
+function componentToImport(path: string, component: UtopiaJSXComponent): ComponentToImport {
+  return {
+    path: path,
+    component: component,
+  }
+}
+
+export function addStoryboardFileToProject(editorModel: EditorModel): EditorModel | null {
+  const storyboardFile = getContentsTreeFileFromString(
+    editorModel.projectContents,
+    StoryboardFilePath,
+  )
+  if (storyboardFile == null) {
+    let namedComponentToImport: ComponentToImport | null = null as any
```

Comment:
Why the `as any` casts here?

Reply:
Because otherwise TypeScript inferred the real type of the two fields as `null`, which then means the else clause of the null check thinks the type is `never` and nothing works.

Reply:
Oh I remember you saying about this. Urgh.